scenario/constraints/NativeTokenConstraint.ts

```diff
@@ -9,7 +9,7 @@ export class NativeTokenConstraint<T extends CometContext, R extends Requirement
       async function (ctx: T): Promise<T> {
         for (const symbol in ctx.assets) {
           const contract = await ctx.world.deploymentManager.contract(symbol);
-          if (contract.deposit) {
+          if (contract['deposit()']) {
```

Comment:
Polygon USDC has a `deposit` function, but it takes multiple arguments:

https://polygonscan.com/address/0xdd9185db084f5c4fff3b4f70e7ba62123b812226#code

Updating this to check for the deposit with the function signature we expect

Reply:
🧼